frontend/components/ActivityDetails/InstallDetails/AppInstallDetails/AppInstallDetails.tsx

```diff
@@ -173,73 +231,104 @@ export const AppInstallDetails = ({
       );
     }
 
-    if (error?.status === 401) {
+    if (errorVPPCommandResult?.status === 401) {
       return deviceAuthToken ? (
         <DeviceUserError />
       ) : (
         <DataError description="Close this modal and try again." />
       );
     }
-  } else if (!result) {
+  } else if (!vppCommandResult) {
     // FIXME: It's currently possible that the command results API response is empty for pending
     // commands. As a temporary workaround to handle this case, we'll ignore the empty response and
     // display some minimal pending UI. This should be updated once the API response is fixed.
   }
 
-  const displayStatus = (status as SoftwareInstallStatus) || "pending";
+  const displayStatus =
+    (fleetInstallStatus as SoftwareInstallStatus) || "pending_install";
   const iconName = INSTALL_DETAILS_STATUS_ICONS[displayStatus];
 
   // Note: We need to reconcile status values from two different sources. From props, we
-  // get the status from the activity item details (which can be "failed", "pending", or
+  // get the status of the Fleet install operation (which can be "failed", "pending", or
   // "installed"). From the command results API response, we also receive the raw status
   // from the MDM protocol, e.g., "NotNow" or "Acknowledged". We need to display some special
   // messaging for the "NotNow" status, which otherwise would be treated as "pending".
-  const isStatusNotNow = result?.status === "NotNow";
-  const isStatusAcknowledged = result?.status === "Acknowledged";
+  const isMDMStatusNotNow = vppCommandResult?.status === "NotNow";
+  const isMDMStatusAcknowledged = vppCommandResult?.status === "Acknowledged";
+
+  const excludeVersions =
+    !deviceAuthToken &&
+    ["pending_install", "failed_install"].includes(fleetInstallStatus);
+
+  const isInstalledByFleet = hostSoftware
+    ? !!hostSoftware.app_store_app?.last_install
+    : true; // if no hostSoftware passed in, can assume this is the activity feed, meaning this can only refer to a Fleet-handled install
 
   const statusMessage = getStatusMessage({
+    isDUP: !!deviceAuthToken,
     displayStatus,
-    isStatusNotNow,
-    isStatusAcknowledged,
-    software_title,
-    host_display_name,
+    isMDMStatusNotNow,
+    isMDMStatusAcknowledged,
+    appName,
+    hostDisplayName,
+    commandUpdatedAt: vppCommandResult?.updated_at || "",
   });
 
-  const showCommandPayload = !!result?.payload;
-  const showCommandResponse =
-    !!result?.result && (isStatusNotNow || status !== "pending");
+  const renderInventoryVersionsSection = () => {
+    if (hostSoftware?.installed_versions?.length) {
+      return <InventoryVersions hostSoftware={hostSoftware} />;
+    }
+    return "If you uninstalled it outside of Fleet it will still show as installed.";
```

Comment:
I'm wondering if this copy is correct, lets sync on this:

For the tables, new behavior is if there is a version returned in hostSoftware?.installed_versions?.length from the latest host fetch, it will still showed as installed

Reply:
This is what was specced. I agree it could be improved, whatever you want to do.

>For the tables, new behavior is if there is a version returned in hostSoftware?.installed_versions?.length from the latest host fetch, it will still showed as installed

I assume you mean "it will show as installed in table's 'status'" row?